includes/api/ApiQueryUserContributions.php

```diff
@@ -225,14 +234,14 @@ private function prepareQuery() {
 				// STRAIGHT_JOIN
 				$tables = array( 'revision', 'recentchanges', 'page' );
 				$this->addOption( 'STRAIGHT_JOIN' );
-				$this->addWhere( 'rc_user_text=rev_user_text' );
+				$this->addWhere( '(rc_user != 0 AND rc_user=rev_user OR rc_user_text=rev_user_text)' );
 				$this->addWhere( 'rc_timestamp=rev_timestamp' );
 				$this->addWhere( 'rc_this_oldid=rev_id' );
 			} else {
 				$tables[] = 'recentchanges';
 				$this->addJoinConds( array( 'recentchanges' => array(
 					'LEFT JOIN', array(
-						'rc_user_text=rev_user_text',
+						'(rc_user != 0 AND rc_user=rev_user OR rc_user_text=rev_user_text)',
```

Comment:
`AND` has precedence over `OR`, so `x AND y OR z` is identical to `(x AND y) OR z` — which is correct in this case, judging by the context; just making sure if that was your intention here.